### DISCUSSION TITLE
An event that's fired if the new tag is a user option?

### DISCUSSION THREAD

**Discussion by igorlanko** [janosh/svelte-multiselect#262] [Ideas]
It might be useful if you want to modify the list of tags by POSTing a request to create one.

In my context, I'd need to differentiate between updating a PGSQL table with new item-tag relation link in a many-to-many table, and creating a new tag in the `tags` table.

What do you think?

**Comment by igorlanko** [comment]
I realize I might've been a little fuzzy about the description here. Basically what I'm curious about is splitting the `on:add` event into two:
- one that's fired when user is selected a tag from the existing list of tags;
- another, when user didn't match any from the existing list and created their own instead.

**Reply by janosh** [reply]
Hey @igorlanko, this feature now exists (forgot to ever report back). The `oncreate` event fires when a user creates a new option (i.e., typed text that doesn't match any existing option), while `onadd` fires for all additions. This lets you differentiate between the two scenarios:

```svelte
<MultiSelect
  options={tags}
  allowUserOptions={true}
  onadd={({ option }) => {
    // Fires for ALL additions - link tag to item in many-to-many table
    await db.item_tags.insert({ item_id, tag_id: option.id })
  }}
  oncreate={async ({ option }) => {
    // Fires ONLY for user-created options - create new tag first
    const newTag = await db.tags.insert({ label: option.label })
    option.id = newTag.id
  }}
/>
```

Note: `oncreate` fires before `onadd`, so you can create the tag record first, then handle the relation in `onadd`.

See the [events demo page](https://multiselect.janosh.dev/events) for a live example and more docs.
